### ISSUE TITLE
Design doc for post-build testing infra

### ISSUE THREAD

**Issue by talkhasib** [magma/magma#9535]

**Comment by talkhasib** [comment]
Include a common report structure

**Comment by uri200** [comment]
Hey @talkhasib @tmdzk , not sure if this belongs here, but few weeks ago I tested a way to access to the labs using an intermediate node under our control (not in github actions). 

```
CircleCI sends API call -> intermediate server with API and OpenVPN with key to the labs (hosted on AWS??) -> ITL Lab
```

Using this solution we were able to run TeraVM from CI (see #8043 to see the results of teramv on CI)

On the intermediate server I just used [shell2http](https://github.com/msoap/shell2http) that provides with a framework to build a simple API to trigger bash commands. I also used OpenConnect to get access to the labs.

Not sure if the solution to this will be use intermediate servers to trigger the jobs. But if this is under consideration maybe it would be good to to take a look at this shell2http. We could use that as POC and then, if we like it, we could build our own API (if needed)





[shell2http](https://github.com/msoap/shell2http)
